rpc-client/src/mock_sender.rs

```diff
@@ -357,6 +357,7 @@ impl RpcSender for MockSender {
                 json!(RpcVersionInfo {
                     solana_core: version.to_string(),
                     feature_set: Some(version.feature_set),
+                    commit: Some(format!("{:x}", version.commit)),
```

Comment:
```suggestion
                    commit: Some(format!("{:08x}", version.commit)),
```

otherwise it'll truncate leading zeros. might prefer to use a [custom serializer](https://serde.rs/field-attrs.html#serialize_with) so we can retain the type-safety of `u32` rather than relenting to `String`